File: Nodejs + Flutter/AppFitness/frontend/lib/models/user.model.dart
class User {
  var id;
  var name;
  var email;
  var password;

  User({this.id, this.name, this.email, this.password});

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
        id: json['_id'],
        name: json['nombre'],
        email: json['email'],
        password: json['password']);
  }
}

File: Nodejs + Flutter/AppFitness/backend/src/models/user.model.js
const { Schema, model } = require('mongoose');
const bcrypt = require('bcryptjs');
const userSchema = new Schema({
    nombre: String,
    email: String,
    password: String,
})
userSchema.methods.hashPass = async(password) => {
    const salt = await bcrypt.genSalt(10)
    const hash = bcrypt.hash(password, salt)
    return hash
};
userSchema.methods.matchPass = async function(password) {
    return await bcrypt.compare(password, this.password)
}

module.exports = model('User', userSchema);